test/unit/extract/xbrl_test.py

```diff
@@ -0,0 +1,192 @@
+                "id4.xbrl": io.BytesIO(b"filer 2 fake filing."),
+            },
+            {
+                "id1": b"filer 1 fake filing.",
+                "id4": b"filer 2 fake filing.",
+            },
+        ),
+    ],
+)
+def test_ferc_xbrl_datastore_get_filings(mocker, file_map, selected_filings):
+    """Test FercXbrlDatastore class."""
+    datastore_mock = mocker.MagicMock()
+
+    # Get mock of archive zipfile
+    archive_mock = datastore_mock.get_zipfile_resource.return_value
+    archive_mock.open.side_effect = file_map.get
+
+    # Call method
+    ferc_datastore = FercXbrlDatastore(datastore_mock)
+    filings = ferc_datastore.get_filings(2021, XbrlFormNumber.FORM1)
+
+    # Check that get_zipfile_resource was called correctly
+    datastore_mock.get_zipfile_resource.assert_called_with(
+        "ferc1", year=2021, data_format="XBRL"
+    )
+
+    # Loop through filings and verify the contents
+    for filing in filings:
+        assert filing.name in selected_filings
+        assert filing.file.getvalue() == selected_filings[filing.name]
+
+
+@pytest.mark.parametrize(
+    "settings,forms",
+    [
+        (
+            FercToSqliteSettings(
+                ferc1_xbrl_to_sqlite_settings=Ferc1XbrlToSqliteSettings(),
+                ferc2_xbrl_to_sqlite_settings=Ferc2XbrlToSqliteSettings(),
+                ferc6_xbrl_to_sqlite_settings=Ferc6XbrlToSqliteSettings(),
+                ferc60_xbrl_to_sqlite_settings=Ferc60XbrlToSqliteSettings(),
+                ferc714_xbrl_to_sqlite_settings=Ferc714XbrlToSqliteSettings(),
+            ),
+            [form for form in XbrlFormNumber],
+        ),
+        (
+            FercToSqliteSettings(
+                ferc1_xbrl_to_sqlite_settings=None,
+                ferc2_xbrl_to_sqlite_settings=Ferc2XbrlToSqliteSettings(),
+                ferc6_xbrl_to_sqlite_settings=Ferc6XbrlToSqliteSettings(),
+                ferc60_xbrl_to_sqlite_settings=Ferc60XbrlToSqliteSettings(),
+                ferc714_xbrl_to_sqlite_settings=Ferc714XbrlToSqliteSettings(),
+            ),
+            [form for form in XbrlFormNumber if form != XbrlFormNumber.FORM1],
+        ),
+        (FercToSqliteSettings(), []),
+    ],
+)
+def test_xbrl2sqlite(settings, forms, mocker):
+    """Test xbrl2sqlite function."""
+    convert_form_mock = mocker.MagicMock()
+    mocker.patch("pudl.extract.xbrl.convert_form", new=convert_form_mock)
+
+    mocker.patch("pudl.extract.xbrl._get_sqlite_engine", return_value="sqlite_engine")
+
+    # Mock datastore object to allow comparison
+    mocker.patch("pudl.extract.xbrl.FercXbrlDatastore", return_value="datastore")
+
+    xbrl2sqlite(
+        ferc_to_sqlite_settings=settings,
+        pudl_settings="pudl_settings",
+        clobber=True,
+        datastore="datastore",
+        batch_size=20,
+        workers=10,
+    )
+
+    if len(forms) == 0:
+        convert_form_mock.assert_not_called()
+
+    for form in forms:
+        if form != XbrlFormNumber.FORM714:
```

Comment:
Ooops this was something I put in for debugging because `assert_any_call` was giving me bad error messages that were easier to parse when only testing the 714 call for reasons that are hard to explain via text and probably not that important haha. Removed now.